sparsity/io.py

```diff
@@ -28,38 +53,45 @@ def traildb_to_coo(db, fieldname):
         sparse.coo_matrix((np.ones(num_events), (r_idx, c_idx)))
 
 
-def to_npz(sf, filename, block_size=None):
+def to_npz(sf, filename, block_size=None, storage_options=None):
     data = _csr_to_dict(sf.data)
     data['metadata'] = \
         {'multiindex': True if isinstance(sf.index, pd.MultiIndex) else False}
     data['frame_index'] = sf.index.values
     data['frame_columns'] = sf.columns.values
     if not filename.endswith('.npz'):
         filename += '.npz'
-    if not filename.startswith('s3://'):
+
+    protocol = urlparse(filename).scheme or 'file'
+    if protocol == 'file':
         fp = open(filename, 'wb')
         np.savez(fp, **data)
     else:
-        _save_npz_s3(data, filename, block_size)
-
-
-def _save_npz_s3(data, filename, block_size=None):
-    if block_size is None:
-        block_size = 2**20 * 100  # 100 MB
-    buffer = BytesIO()
-    np.savez(buffer, **data)
-    buffer.seek(0)
-    fs = S3FileSystem()
-    with fs.open(filename, 'wb', block_size) as s3f:
+        if block_size is None:
+            block_size = 2 ** 20 * 100  # 100 MB
+        buffer = BytesIO()
+        np.savez(buffer, **data)
+        buffer.seek(0)
+        _save_remote(buffer, filename, block_size, storage_options)
+
+
+def _save_remote(buffer, filename, block_size=None, storage_options=None):
+    if storage_options is None:
+        storage_options = {}
+    protocol = urlparse(filename).scheme or 'file'
```

Comment:
`scheme` shouldn't be `'file'` here. I think leaving this as an option is misleading.

Reply:
that's correct